public/project/javascripts/controllers/NavController.js

```diff
@@ -1,23 +1,55 @@
 app.controller('NavController', [
-	'$scope',
-	'$rootScope',
-	'$state',
-	function(
-		$rootScope,
-		$scope,
-		$state) {
-	'use strict';
-	console.log('#### Nav Controller');
-	$scope.goToDesktopRoute = function(route) {
-		if(route === 'home') {
-			$state.go('app.v1.landing');
-		}
-		if(route === 'music') {
-			$state.go('app.v1.music-desktop');
-		}
-		if(route === 'booking') {
-			$state.go('app.v1.booking-desktop');
-		}
-		$rootScope.$broadcast('hide sidenav');
-	}
-}]);
+  '$scope',
+  '$rootScope',
+  '$state',
+  '$timeout',
+  'StateService',
+  function(
+    $scope,
+    $rootScope,
+    $state,
+    $timeout,
+    StateService) {
+
+    'use strict';
+
+    console.log('#### Nav Controller');
+
+    $rootScope.$on('toggleSidebar', toggleSidebar);
+
+    $scope.navigate = function(route) {
+      if (route === 'home') {
+        $state.go('app.v1.landing');
+      } else if (route === 'music') {
+        $state.go('app.v1.music-desktop');
+      } else if (route === 'booking') {
+        $state.go('app.v1.booking-desktop');
+      } else if (route === 'cars') {
+        console.log('navigate to cars');
+      } else {
+        console.error('route not found');
+      }
+    };
```

Comment:
This is cool. But it would be faster and less code if you passed the actual route instead of a nickname for the route. For example, instead of passing home, just pass 'app.v1.landing'. Then you can get rid of all the conditionals. That is as dynamic as you can make this, since there is no point hard coding 'home' aaaand 'app.v1.landing'